### PR TITLE
VPlan/PatternMatch: introduce m_c_Mul (NFC)

### PR DESCRIPTION
**Pull request by artagnon**
Introduce a commutative version of m_Mul, and simplify a usage based on it.

### CODE REVIEW

File: llvm/lib/Transforms/Vectorize/VPlanPatternMatch.h [hunk 1]
```diff
@@ -284,11 +290,10 @@ m_BinaryOr(const Op0_t &Op0, const Op1_t &Op1) {
   return m_Binary<Instruction::Or, Op0_t, Op1_t, Commutative>(Op0, Op1);
 }
 
-template <typename Op0_t, typename Op1_t>
-inline AllBinaryRecipe_match<Op0_t, Op1_t, Instruction::Or,
-                             /*Commutative*/ true>
+template <typename Op0_t, typename Op1_t, bool Commutative = true>
+inline AllBinaryRecipe_match<Op0_t, Op1_t, Instruction::Or, Commutative>
```
Comment (fhahn) [inline llvm/lib/Transforms/Vectorize/VPlanPatternMatch.h:294]: change unrelated to adding `m_c_Mul`?

**Comment by artagnon** [comment]
Addressed review.

**Review by fhahn — APPROVED** [review]
LGTM, thanks!

**Comment by ayalz** [comment]
Mentioned in https://github.com/llvm/llvm-project/pull/92539#discussion_r1606051541

Any test(s) to exercise?

**Comment by artagnon** [comment]
This codepath is quite well-tested:

```
  LLVM :: Transforms/LoopVectorize/ARM/mve-gather-scatter-tailpred.ll
  LLVM :: Transforms/LoopVectorize/RISCV/strided-accesses.ll
  LLVM :: Transforms/LoopVectorize/runtime-check-needed-but-empty.ll
  LLVM :: Transforms/LoopVectorize/runtime-checks-hoist.ll
  LLVM :: Transforms/LoopVectorize/select-reduction.ll
  LLVM :: Transforms/LoopVectorize/version-stride-with-integer-casts.ll
```

**Pull request merged**